main/src/com/google/refine/model/recon/StandardReconConfig.java

```diff
@@ -217,12 +223,26 @@ public ReconType getReconType() {
         }
         return null;
     }
+        @Override
+        @JsonIgnore
+        public int getBatchSize (Project project) {
```

Comment:
Instead of taking the entire `Project` as argument, perhaps you could pass only the number of rows, as this is all we need here? This would make sure that other implementations of `ReconConfig` also don't rely on too exotic aspects of the project to determine their batch size.